docs/content/filtering.md

```diff
@@ -5,31 +5,65 @@ description: "Filtering, includes and excludes"
 
-Rclone has a sophisticated set of include and exclude rules. Some of
-these are based on patterns and some on other things like file size.
+Filter flags determine which files rclone`sync`, `move`, `ls`, `lsl`,
+`md5sum`, `sha1sum`, `size`, `delete` and `check` commands are applied
+to.
 
-The filters are applied for the `copy`, `sync`, `move`, `ls`, `lsl`,
-`md5sum`, `sha1sum`, `size`, `delete` and `check` operations.
-Note that `purge` does not obey the filters.
+They are specified in terms of path/file name
+patterns; path/file lists; file age and size, or presence of
+a file in a directory. Bucket remotes without the concept of
+directory apply filters to object key, age and size in an
+analogous way.
 
-Each path as it passes through rclone is matched against the include
-and exclude rules like `--include`, `--exclude`, `--include-from`,
-`--exclude-from`, `--filter`, or `--filter-from`. The simplest way to
-try them out is using the `ls` command, or `--dry-run` together with
-`-v`. `--filter-from`, `--exclude-from`, `--include-from`, `--files-from`,
-`--files-from-raw` understand `-` as a file name to mean read from standard
-input.
+Rclone `purge` does not obey filters.
 
-## Patterns ##
+To test filters without risk of damage to data, apply them to `rclone
+ls`, or with the `--dry-run` and `-vv` flags.
 
-The patterns used to match files for inclusion or exclusion are based
-on "file globs" as used by the unix shell.
+Rclone filter patterns can only be used in filter command line options, not
+in the specification of a remote.
 
-If the pattern starts with a `/` then it only matches at the top level
-of the directory tree, **relative to the root of the remote** (not
-necessarily the root of the local drive). If it doesn't start with `/`
-then it is matched starting at the **end of the path**, but it will
-only match a complete path element:
+Eg `rclone copy "remote:dir*.jpg" /path/to/dir` does not have a filter effect.
+`rclone copy remote:dir /path/to/dir --include "*.jpg"` does.
+
+**Important** Avoid mixing any two of `--include...`, `--exclude...` or
+`--filter...` flags in an rclone command. The results may not be what
+you expect. Instead use a `--filter...` flag.
+
+## Patterns for matching path/file names ##
+
+### Pattern syntax ###
+
+Rclone's matching rules follow a glob style:
+
+    `*`         matches any sequence of non-separator (`/`) characters
+    `**`        matches any sequence of characters
+    `?`         matches any single non-separator (`/`) character
+    `[` [ `!` ] { character-range } `]`
+                character class (must be non-empty)
+    `{` pattern-list `}`
+                pattern alternatives
+    c           matches character c (c != `*`, `**`, `?`, `\`, `[`, `{`, `}`)
+    `\` c       matches character c
+
+character-range:
+
+    c           matches character c (c != `\\`, `-`, `]`)
+    `\` c       matches character c
+    lo `-` hi   matches character c for lo <= c <= hi
```

Comment:
The character ranges are actually a bit more powerful than this - they are the character ranges from go regular expressions - https://golang.org/pkg/regexp/syntax/ - perhaps mention that somewhere but no need to go into detail!

Reply:
Will look again at that, thanks.

Reply:
OK, I'm not sure I understand. Rclone filters are glob based and are transformed internally and applied by rclone, when deciding which files to apply an rclone command to, in a regex form. The syntax presented is the classic glob * for a character and the regex is classic, well perl extended regex, where * says repeat the prior char one or more times.

I'm not sure whether your point applies to the regex syntax on https://golang.org/pkg/regexp/syntax/ in general or just to ranges. I can't see that there is anything directly more complex there for ranges that would be possible to apply through a glob syntax.

Are you saying the following style syntax can be applied in a range within a file name filter?

```
[[:alnum:]]    alphanumeric (== [0-9A-Za-z])
[[:alpha:]]    alphabetic (== [A-Za-z])
[[:ascii:]]    ASCII (== [\x00-\x7F])
[[:blank:]]    blank (== [\t ])
[[:cntrl:]]    control (== [\x00-\x1F\x7F])
[[:digit:]]    digits (== [0-9])
[[:graph:]]    graphical (== [!-~] == [A-Za-z0-9!"#$%&'()*+,\-./:;<=>?@[\\\]^_`{|}~])
[[:lower:]]    lower case (== [a-z])
[[:print:]]    printable (== [ -~] == [ [:graph:]])
[[:punct:]]    punctuation (== [!-/:-@[-`{-~])
[[:space:]]    whitespace (== [\t\n\v\f\r ])
[[:upper:]]    upper case (== [A-Z])
[[:word:]]     word characters (== [0-9A-Za-z_])
[[:xdigit:]]   hex digit (== [0-9A-Fa-f])
```
The reason I am pursuing this is because later in the page I say there isn't a way for the user to input rclone's regex internal filter syntax directly and instead provide the regex syntax link as an aid to comprehension of `--dump filters` output. Maybe what I say there isn't true? 

>